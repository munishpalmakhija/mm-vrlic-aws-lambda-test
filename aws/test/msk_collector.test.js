/*
Copyright 2018 VMware, Inc.
SPDX-License-Identifier: MIT
*/
const { lintTestEnv } = require('./helper.test');
const { mskSampleData } = require('./sample_files/msk_testdata');

const {
  MSKHttpCollector,
} = require('../src/index');


describe('MSKHttpCollector', () => {
  const collector = new MSKHttpCollector(lintTestEnv);

  describe('processLogsJson', () => {
    it('should process logs JSON', () => {
      const logsJson = mskSampleData();
      collector.processLogsJson(logsJson);
      expect(logsJson).toMatchSnapshot();
    });
  });
});

