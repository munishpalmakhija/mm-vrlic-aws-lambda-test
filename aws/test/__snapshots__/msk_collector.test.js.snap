// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MSKHttpCollector processLogsJson should process logs JSON 1`] = `
Object {
  "Records": Array [
    Object {
      "eventSource": "aws:kafka",
      "eventSourceArn": "arn:aws:kafka:us-east-2:827066587692:cluster/MyCluster/1cbf212b-3937-4536-a3b0-c09868d7e5f5-4",
      "ingest_timestamp": 1538769915817,
      "log_type": "aws_msk",
      "offset": 6,
      "partition": 0,
      "timestamp": 1601289409497,
      "timestampType": "CREATE_TIME",
      "topic": "mytopic",
      "value": "Message #7",
    },
  ],
}
`;
